xbmc/pvr/providers/PVRProvider.h

```diff
@@ -0,0 +1,217 @@
+   */
+  bool SetIconPath(const std::string& strIconPath);
+
+  /*!
+   * @brief Get this provider's country codes (ISO 3166). Separated by
+   *        PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return This provider's country codes.
+   */
+  std::string GetCountry() const;
+
+  /*!
+   * @brief Set the country codes for this provider
+   * @param strCountry The new ISO 3166 country codes for this provider. Separated by
+   *                   PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return true if the country codes were updated successfully
+   */
+  bool SetCountry(const std::string& strCountry);
+
+  /*!
+   * @brief Get this provider's language codes (RFC 5646). Separated by
+   *        PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return This provider's language codes
+   */
+  std::string GetLanguage() const;
+
+  /*!
+   * @brief Set the language codes for this provider
+   * @param strLanguage The new RFC 5646 language codes for this provider. Separated by
+   *                    PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return true if the language codes were updated successfully
+   */
+  bool SetLanguage(const std::string& strLanguage);
+
+  /*!
+   * @brief Get if this provider's has thumb image path. Note only PVR add-on providers.
+   * @return True is this add-on provider has a thumb image path, false otherwise.
+   */
+  bool HasThumbPath() const;
+
+  /*!
+   * @brief Get this provider's thumb image path. Note only PVR add-on providers will set this value.
+   * @return This add-on provider's thumb image path.
+   */
+  std::string GetThumbPath() const { return m_strThumbPath; }
+
+  /*!
+   * @brief Whether a provider is a default provider of a PVR Client add-on or not
+   * @return True is this provider is of a PVR Client add-on, false otherwise.
+   */
+  bool IsClientProvider() const { return m_bIsClientProvider; }
+
+  /*!
+   * @brief updates this provider from the provided entry
+   * @param fromProvider A provider containing the data that shall be merged into this provider's data.
+   * @param updateMode update as User, Client or DB
+   * @return true if the provider was updated successfully
+   */
+  bool UpdateEntry(const std::shared_ptr<CPVRProvider>& fromProvider,
+                   ProviderUpdateMode updateMode);
+
+  /*!
+   * @brief Persist this provider in the local database.
+   * @param updateRecord True if an existing record should be updated, false for an insert
+   * @return True on success, false otherwise.
+   */
+  bool Persist(bool updateRecord = false);
+
+  /*!
+   * @brief Delete this provider from the local database.
+   * @return True on success, false otherwise.
+   */
+  bool DeleteFromDatabase();
+
+  mutable CCriticalSection m_critSection;
+
+private:
+  CPVRProvider(const CPVRProvider& provider) = delete;
+  CPVRProvider& operator=(const CPVRProvider& orig) = delete;
+
+  int m_iProviderId; /*!< the identifier given to this provider by the TV database */
```

Comment:
Any reason not to default initialize this and (other) member(s) here?

Reply:
Nope, done